reco_utils/dataset/sparse.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+'''
+Generate the user/item affinity matrix from a pandas dataframe and vice versa
+'''
+
+import pandas as pd
+
+import numpy as np
+import random
+import math
+
+import itertools
+
+from scipy.sparse import coo_matrix
+import logging
+
+#import default parameters
+from reco_utils.common.constants import (
+    DEFAULT_USER_COL,
+    DEFAULT_ITEM_COL,
+    DEFAULT_RATING_COL,
+    DEFAULT_TIMESTAMP_COL,
+    PREDICTION_COL,
+)
+
+#for logging
+log = logging.getLogger(__name__)
+
+
+class affinity_matrix:
+    '''
+
+    Args:
+        DF (pandas.DataFrame): a dataframe containing the data
+        col_user: default name for user column
+        col_item: default name for item column
+        col_rating: default name for rating columns
+        col_time: default name for timestamp columns
+        save_model: if True it saves the item/user maps
+        save_path: default path to save item/user maps
+
+    '''
+
+    #initialize class parameters
+    def __init__(
+        self,
+        DF,
+        col_user = DEFAULT_USER_COL,
+        col_item = DEFAULT_ITEM_COL,
+        col_rating = DEFAULT_RATING_COL,
+        col_time = DEFAULT_TIMESTAMP_COL,
+        save_model = False,
+        save_path = 'saver',
+        debug = False,
+    ):
+
+        self.df = DF #dataframe
+
+        #pandas DF parameters
+        self.col_rating = col_rating
+        self.col_item = col_item
+        self.col_user = col_user
+
+        #Options to save the model for future use
+        self.save_model_ = save_model

```

Comment:
Not sure why the extra `_` in `self.save_model_`, we can do directly `self.save_model`. 
Optionally, a trick to do what you want to do, is to have just one variable, `save_path`, initialized to None. Then you do:
```
if self.save_path is not None:
     n p.save(...)
``` 
this way you are saving one variable and reducing complexity

Reply:
good tips, thanks!

Reply:
good idea, thanks !